src/main/java/com/uber/cadence/worker/Worker.java

```diff
@@ -0,0 +1,236 @@
+        Objects.requireNonNull(taskList, "taskList");
+        if (options == null) {
+            options = new WorkerOptions.Builder().build();
+        }
+        this.options = options;
+        SingleWorkerOptions activityOptions = toActivityOptions(options);
+        if (!options.isDisableActivityWorker()) {
+            activityWorker = new SyncActivityWorker(service, domain, taskList, activityOptions);
+        } else {
+            activityWorker = null;
+        }
+        SingleWorkerOptions workflowOptions = toWorkflowOptions(options);
+        if (!options.isDisableWorkflowWorker()) {
+            workflowWorker = new SyncWorkflowWorker(service, domain, taskList, workflowOptions);
+        } else {
+            workflowWorker = null;
+        }
+    }
+
+    private SingleWorkerOptions toActivityOptions(WorkerOptions options) {
+        return new SingleWorkerOptions.Builder()
+                .setDataConverter(options.getDataConverter())
+                .setIdentity(options.getIdentity())
+                .setPollerOptions(options.getActivityPollerOptions())
+                .setReportCompletionRetryOptions(options.getReportActivityCompletionRetryOptions())
+                .setReportFailureRetryOptions(options.getReportActivityFailureRetryOptions())
+                .setTaskExecutorThreadPoolSize(options.getMaxConcurrentActivityExecutionSize())
+                .build();
+    }
+
+    private SingleWorkerOptions toWorkflowOptions(WorkerOptions options) {
+        return new SingleWorkerOptions.Builder()
+                .setDataConverter(options.getDataConverter())
+                .setIdentity(options.getIdentity())
+                .setPollerOptions(options.getWorkflowPollerOptions())
+                .setReportCompletionRetryOptions(options.getReportWorkflowCompletionRetryOptions())
+                .setReportFailureRetryOptions(options.getReportWorkflowFailureRetryOptions())
+                .setTaskExecutorThreadPoolSize(options.getMaxConcurrentWorklfowExecutionSize())
+                .build();
+    }
+
+    /**
+     * Register workflow implementation classes with a worker.
+     * A workflow implementation class must implement at least one interface with a method annotated with
+     * {@link com.uber.cadence.workflow.WorkflowMethod}. That method becomes a workflow type
+     * that this worker supports.
+     * <p>
+     * Implementations that share a worker must implement different interfaces as a workflow type
+     * is identified by the workflow interface, not by the implementation.
+     * </p><p>
+     * The reason for registration accepting workflow class, but not the workflow instance is
+     * that workflows are stateful and a new instance is created for each workflow execution.
+     * </p>
+     */
+    public void registerWorkflowImplementationTypes(Class<?>... workflowImplementationClasses) {
+        workflowWorker.setWorkflowImplementationTypes(workflowImplementationClasses);
```

Comment:
we check activityWorker == null in registerActivitiesImplementations, should we do that for workflowWorker here?

Reply:
and checkStarted() as well?

Reply:
good catch